mantidproject/mantid
# Rewrite SaveOpenGenieAscii to save valid files

Description of work.
This PR rewrites `SaveOpenGenieAscii` to both clean it up and fix the format for ENGIN-X OpenGenie. At a later date a rarely used procedure could need a new parameter in the output file but at the moment this PR covers those needed in their `analyze_scan` routine which they need for some data processing.

Additionally it adds a unit test which just checks the output against a reference file. 

**To test:**
As a copy of ENGIN-X routines is difficult to setup and test these files with you will have to take my word the file (currently) works correctly in OpenGenie. 
- Code Review
- Run the unit test

Fixes #18084 and #18443 .

[Release notes](https://github.com/mantidproject/mantid/pull/18468/files#diff-2abf582e699070b9e379ddfafa4d1094)

~Additionally please do **not** merge this branch yet. There is a issue open with Github about the fact the author appears twice in compare commits using Git 2.11 that links to this branch. If the branch is merged the commits will not longer appear in the comparison.~

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
